src/Microsoft.Health.Fhir.Azure/ContainerRegistry/ContainerRegistryBasicTokenProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Operations.DataConvert;
+using Microsoft.Health.Fhir.Core.Features.Operations.DataConvert.Models;
+
+namespace Microsoft.Health.Fhir.Azure.ContainerRegistry
+{
+    /// <summary>
+    /// Implement basic auth logic using ACR admin keys.
+    /// Reference: https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md#catalog-listing-using-spadmin-with-basic-auth
+    /// </summary>
+    public class ContainerRegistryBasicTokenProvider : IContainerRegistryTokenProvider
+    {
+        private readonly DataConvertConfiguration _dataConvertConfiguration;
+
+        public ContainerRegistryBasicTokenProvider(IOptions<DataConvertConfiguration> dataConvertConfiguration)
+        {
+            EnsureArg.IsNotNull(dataConvertConfiguration, nameof(dataConvertConfiguration));
+
+            _dataConvertConfiguration = dataConvertConfiguration.Value;
+        }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
```

Comment:
Why make this method async if we don't actually await anything?

Reply:
Because I want to make the method async in the interface since we will have an async way to obtain token in the PaaS service. But in OSS, we don't need do it in async way.

Reply:
you can try to use Task.FromResult

Reply:
I'll double that.
Interface doesn't actually specify is it async or not, it can't do that, what it does it specify what return value is Task.
To simplify you can do this:
```
public Task<string> Ham(){
 var str = "spam";
 return Task.FromResult(str).
}
```

Reply:
Sure. Will change to return a Task. Thanks!